frontend/dockerfile/dockerfile2llb/directives.go

```diff
@@ -2,16 +2,30 @@ package dockerfile2llb
 
 import (
 	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
 	"io"
 	"regexp"
 	"strings"
+	"sync"
 
 	"github.com/moby/buildkit/frontend/dockerfile/parser"
 )
 
 const keySyntax = "syntax"
 
-var reDirective = regexp.MustCompile(`^#\s*([a-zA-Z][a-zA-Z0-9]*)\s*=\s*(.+?)\s*$`)
+type directiveParser interface {
+	parse(r io.Reader) map[string]Directive
+}
+
+var defaultParser = commentDirectiveParser("#")
+
+var parsers = []directiveParser{
+	allowShebang(defaultParser),
+	commentDirectiveParser("//"),
```

Comment:
Why doesn't the comment parser allow shebang? Or maybe it makes more sense just to strip shebang always on `DetectSyntax`

Reply:
The logic was that a language that doesn't support `#` as comments likely wouldn't support shebangs (which starts with the `#` comment symbol).

But agreed, I'm not sure we want to make that judgement. I'll add a change to just wrap the comment directive parser as well, since handling it in `DetectSyntax` would mean that we'd also allow shebangs for JSON as well, which would mean the whole file wouldn't stay JSON syntax.